internal/util/kubernetes/kubernetes_test.go

```diff
@@ -154,3 +368,67 @@ func TestIsValidKubernetesManifestFile(t *testing.T) {
 	}
 
 }
+
+func TestApplyOwnerShipReferenceOnlyIfNotExisting(t *testing.T) {
+	resource := `apiVersion: v1
+kind: Pod
+metadata:
+  name: my-custom-resource
+  ownerReferences:
+  - apiVersion: apps/v1
+    kind: Deployment
+    name: my-deployment
+    uid: <uid-of-my-deployment>
+    controller: false
+    blockOwnerDeletion: true
+  - apiVersion: v1
+    kind: ConfigMap
+    name: my-configmap
+    uid: <uid-of-my-configmap>
+    controller: false
+    blockOwnerDeletion: true
+  - apiVersion: v1
+    kind: ConfigMap
+    name: my-configmap
+    uid: awew
+    controller: false
+    blockOwnerDeletion: true`
+
+	gitsync := &v1alpha1.GitSync{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "GitSync",
+			APIVersion: "1",
+		},
+		ObjectMeta: metav1.ObjectMeta{Name: "gitsync-test", UID: "awew"},
+		Spec:       v1alpha1.GitSyncSpec{},
+		Status:     v1alpha1.GitSyncStatus{},
+	}
+	reference, err := ApplyGitSyncOwnership(resource, gitsync)
+	assert.NoError(t, err)
+	log.Println(err)
+	assert.Equal(t, `apiVersion: v1
+kind: Pod
+metadata:
+  name: my-custom-resource
+  ownerReferences:
+  - apiVersion: apps/v1
+    blockOwnerDeletion: true
+    controller: false
+    kind: Deployment
+    name: my-deployment
+    uid: <uid-of-my-deployment>
+  - apiVersion: v1
+    blockOwnerDeletion: true
+    controller: false
+    kind: ConfigMap
+    name: my-configmap
+    uid: <uid-of-my-configmap>
+  - apiVersion: v1
+    blockOwnerDeletion: true
+    controller: false
```

Comment:
I don't quite follow why controller can be false in the case you explicitly set to true at https://github.com/numaproj-labs/numaplane/pull/143/files#diff-3d19777dfd8078f46a7e12b70830cbbb7f5855463b120852fd92c0a30120ffa5R124

Reply:
Hmm, actually are you testing if the uid already exist then it will not be added again? If so, the test name should be updated to `TestApplyOwnerShipReferenceAlreadyExists`, right?

Reply:
> I don't quite follow why controller can be false in the case you explicitly set to true at

it was just unit tests ,so i dint much focus on the values ,as I am just testing whether the ownership get appended or not

Reply:
> I saw the new changes didn't take the suggestion, but I'm good with it as I don't feel strong about this.

I have used  meta v1 unstructured as suggested by you [here]( https://github.com/numaproj-labs/numaplane/blob/54f447bbcabb7285efd8b452253d11ae397aa9f4/internal/util/kubernetes/kubernetes.go#L109) did I miss something ? @xdevxy